linfa-bayes/examples/winequality.rs

```diff
@@ -0,0 +1,109 @@
+#[macro_use]
+extern crate ndarray;
+
+use std::error::Error;
+use std::fs::File;
+
+use csv::ReaderBuilder;
+use flate2::read::GzDecoder;
+use ndarray::{Array1, Array2, Axis};
+use ndarray_csv::Array2Reader;
+
+use linfa::metrics::IntoConfusionMatrix;
+use linfa_bayes::{FittedGaussianNb, GaussianNb};
+
+fn main() -> Result<(), Box<dyn Error>> {
+    // Read in the wine-quality dataset from dataset path
+    // The `.csv` data is two dimensional: Axis(0) denotes y-axis (rows), Axis(1) denotes x-axis (columns)
+    let dataset = read_array("../datasets/winequality-red.csv.gz")?;
+
+    // Training data goes along the first axis (rows)
+    let npoints = dataset.len_of(Axis(0));
+    // Take 90% of the dataset as training data, and the remaining 10% as validation data
+    // `floor` is used here to explicitely down-round the number
+    let ntrain = (npoints as f32 * 0.9).floor() as usize;
```

Comment:
can you use the new dataset interface for the splitting here, here is an example: https://github.com/rust-ml/linfa/blob/master/linfa-svm/examples/winequality.rs

Reply:
Some errors like `BayesError::Priors` are very specific to the algorithm, do you still want to move them over to the root?

Reply:
The `split_with_ratio` method for `Dataset` returns a slice for the target attribute, now should I,

- convert slice to ndarray and proceed? or
- make the algorithm work with a slice target?

The second option will make linfa more ergonomic to use, but I want to know your thoughts

Reply:
yeah the second would be more useful

Reply:
The example now used `split_with_ratio` method